dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/factory.py

```diff
@@ -1977,6 +2028,149 @@ def prepare_for_evalfadept_external_call(self, used_functions):
+        if len(called_func) > 0:
+            # filter whatever is assigned in this line
+            line_split_by_equal = line.split('=')
+            assert(len(line_split_by_equal) >= 2)
+            call_line = line_split_by_equal[0] + " = "
+            func_call_line = line_split_by_equal[1]
+
+            # Split this line at each function call ('(') and parameter (',')
+            line_split_by_parenthesis = re.split('(\()', func_call_line)
+            line_split = []
+            for l in line_split_by_parenthesis:
+                line_split.extend(l.split(','))
+
+            # stack (FILO) containing the function called sorted by call stack
+            stack_func_called = []
+            # same size as stack_func_called
+            # each index contains the first id of line_split at which the first parameter of the corresponding function is found
+            stack_param_idx_func_called = []
+            idx = 0
+
+            # True if the first function of the stack requires adept inputs, False otherwise
+            main_func_is_adept = False
+
+            while idx < len(line_split):
+                l = line_split[idx].strip()
+
+                #hack to handle the case data->localData[0]->derivativesVars[...] /* der(a) STATE_DER /
+                if l.endswith("/* der"):
+                    idx+=1
+                    l+=line_split[idx].strip()
+                    idx+=1
+                    l+=line_split[idx].strip()
+                #hack to handle cast
+                while l.endswith("modelica_integer)") or l.endswith("modelica_real)") or l.endswith("modelica_boolean)"):
+                    idx+=1
+                    l+=line_split[idx].strip()
+                    idx+=1
+                    while l.endswith("("):
+                        l+=line_split[idx].strip()
+                #Filter empty indexes
+                if len(l) == 0:
+                    idx+=1
+                    continue
+
+                #Put back any opening parenthesis into the final line
+                if l =='(':
+                    call_line+= l
+                    idx+=1
+                    continue
+
+                #Is there a function call in this index?
+                function_found = None
+                for func_name in called_func:
+                    if func_name in l:
+                        function_found = func_name
+                        break
+
+                if function_found is not None:
+                    #First case: there is a function call here.
+                    # Push it on the stack
+                    stack_func_called.append(function_found)
+                    stack_param_idx_func_called.append(0)
+                    idx+=1
+                    # Replace this function by its adept equivalent only if
+                    # - this function has an adept equivalent
+                    # - the main function has an adept equivalent if this function is called as a parameter
```

Comment:
What happens if there is no adept equivalent? Is it possible?

With the latest part at the end (line 2150 - 2170), do we support all the possible cases?

Reply:
Fine for me. Of course, we will see how robust this is (one good point is that we don't use so much function and that is something that will last I believe).